0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md

```diff
@@ -1,62 +1,127 @@
 
-The Simple Payment Setup Protocol (SPSP) is a basic application-layer Interledger protocol for wallet-to-wallet or wallet-to-merchant payments.
+## Preface
 
-## Stages
+This document describes the Simple Payment Setup Protocol (SPSP), a basic application-layer Interledger protocol for exchanging payment information between senders and recipients.
 
-SPSP performs its function in five stages:
+## Introduction 
 
-1. Discovery
-2. Query
-3. Quoting
-4. Setup
-5. Execution
+### Motivation
 
-### Discovery
+The Interledger Protocol does not specify how payment details, such as the ILP Packet or Crypto Condition, should be exchanged between the sender and recipient. SPSP is a minimal protocol that uses HTTP for communicating these details.
 
-Whenever possible, URIs should be exchanged out-of-band and discovery should be skipped. However, in some cases, it may be useful to have a standardized user-friendly identifier. This discovery method describes how to resolve such an identifier to a queryable SPSP endpoint.
+### Scope
 
-First, the sender uses Webfinger ([RFC 7033](https://tools.ietf.org/html/rfc7033)) to look up an identifier (e.g. `bob@red.ilpdemo.org`):
+SPSP provides for exchanging basic payment details needed by a sender to confirm the details of and set up an ILP payment. It is intended for use by end-user applications.
 
-``` http
-GET /.well-known/webfinger?resource=acct%3Abob%40red.ilpdemo.org HTTP/1.1
-Host: red.ilpdemo.org
-Accept: application/json
-```
-``` http
-HTTP/1.1 200 OK
-Content-Type: application/json
+### Interfaces
 
-{
-  "subject": "acct:bob@red.ilpdemo.org",
-  "links": [
-    {
-      "rel": "https://interledger.org/rel/receiver",
-      "href": "https://red.ilpdemo.org/api/receivers/bob"
-    }
-  ]
-}
-```
-[Try this request](https://red.ilpdemo.org/.well-known/webfinger?resource=acct%3Abob%40red.ilpdemo.org)
+SPSP may be used by end-user applications, such as a digital wallet with a user interface for the sender to initiate payments. SPSP clients and receivers use ILP modules to send and receive Interledger payments.
+
+SPSP messages MUST be exchanged over HTTPS.
+
+### Operation
+
+Any SPSP recipient will create a receiving HTTP endpoint called the *receiver*. The sender can query this endpoint to get information about the type of payment that can be made to this receiver. The sender also uses the receiver endpoint to set up the ILP payment.
+
+### Definitions
+
+#### SPSP Client
+
+The sender application that uses SPSP to interact with the SPSP Server.
+
+#### SPSP Server
+
+The server used on the recipient's side to handle SPSP requests.
+
+#### Receiver
+
+The specific HTTP endpoint on the SPSP used for setting up a payment.
+
+## Overview
+
+### Relation to Other Protocols
+
+SPSP is used for exchanging payment information before an ILP payment is initiated. It uses the [Interactive Transport Protocol (ITP)](../0011-interactive-transport-protocol) for generating conditions.
```

Comment:
See https://github.com/interledger/rfcs/pull/76
I'll update this PR so it depends on that other one